BDArmory/Weapons/ModuleWeapon.cs

```diff
@@ -3260,21 +3266,12 @@ bool CanFire(float AmmoPerShot)
             }
             vessel.GetConnectedResourceTotals(AmmoID, out double ammoCurrent, out double ammoMax);
             ammoCount = ammoCurrent;
-            if (ammoCount >= AmmoPerShot)
+            if (ammoCount >= AmmoPerShot * 0.995f) //catch floating point errors from fractional ammo spread across multiple boxes
+            // TODO?? Change code to some sort of list of ammoboxes to only draw from box with current highest resouce amount to do proper integer reductions?
```

Comment:
Proper integer reduction would be the better option, yeah.
However, the resource steal game mode would likely need to be tweaked to not take fractional amounts from parts.

Reply:
There is a integerAmounts option in resource Steal already, might just be a case of having that be regarded as true if the resource being stolen is an ammo resource. Something to test should integer ammo deductions get implemented, but that's something for a future PR.

Reply:
The `integerAmounts` only considers the total amount of the resource available and is being set for ammo and countermeasures (not for fuel), but it doesn't take into account integer amounts in the individual containers.
I agree that that can wait for a future PR though. I'll add it to the TODO list when this one gets merged.